### ISSUE TITLE
Bandwidth utilization 

### ISSUE THREAD

**Issue by JonasAhl** [ietf-ccamp-wg/draft-ietf-ccamp-mw-topo-yang#3]
o	Do nodes provide this info on a link/carrier level?
o	How often do they change?
o	Average value over which period of time?
o	Retrieve from node?

**Comment by DanielaSpreafico** [comment]
I add my considerations about Bandwidth utilization.

The Bandwidth utilization could provide a way to understand whether the nominal (maximum) radio capacity is coherent with the traffic passing through it.
The formula to calculate it has to use the throughput (transmitted data in specific period time ) and the nominal capacity (or nominal maximum bandwidth)
Bandwidth-utilization = [throughput (Mb/s) / nominal-maximum-bandwidth (Mb/s)]  * 100

Is this information is on link level, taking into account the single carriers and the mode they are combined for the link (1+1, 2+0, etc). Or is it on carrier level (each single carrier on a link, whatever the mode of the radio link?

Due to the frequent change of throughput then also the Bandwidth utilization could change frequently.
It is mainly a performance counter (both as statistic/timely data value or on period-time basis data value).

Could it be recovered from the node? It is not a standard counter, so not all devices are able to provide it.
Or based on other data from the node (tx bytes sent) ? In this case it is needed a reference time to calculated delta-time and delta-tx-bytes-sent).

**Comment by JonasAhl** [comment]
I agree with the definition and questions raised by Daniela in the previous comment.
Another question is if it perhaps is more relevant to measure this on the Ethernet link level rather than on radio link level. It is most likely dependent upon the use case to be supported.

**Comment by italobusi** [comment]
FYI, this is the definition of band in [RFC8776](https://datatracker.ietf.org/doc/html/rfc8776):

```
       leaf one-way-utilized-bandwidth {
         type rt-types:bandwidth-ieee-float32;
         units "bytes per second";
         default "0x0p0";
         description
           "Bandwidth utilization that represents the actual
            utilization of the link (i.e., as measured in the router).
            For a bundled link, bandwidth utilization is defined to
            be the sum of the component link bandwidth utilizations.";
       }

```

**Comment by JonasAhl** [comment]
Bandwidth utilization is an important information for use cases such as energy efficiency.
There are a couple of questions to be answered:
- Is there a need to manage this in a multidomain context or not, and in such case do you measure utilization on a radio link level or on e.g. the Ethernet level which is a multidomain technology?
- Is this specific for microwave or is it generic and can be used for many technologies and thereby perhaps should be included in the topology model itself, alternatively included in the bandwidth-availability module.
-

**Comment by JonasAhl** [comment]
Check with authors of RFC 8776 why not in te-topology.

**Comment by JonasAhl** [comment]
Agreed to define a new attribute specifically for microwave topology, instead of reusing the grouping above.
Daniela to suggest a definition of the leaf & description.

**Comment by JonasAhl** [comment]
The period over which the bandwidth is measured should be clear. 1s? 6 sec? Leo to investigate.
Check what the difference is between the leaf defined by RFC 8776 and the leaf actual-bandwidth in the generic module ietf-bandwidth-availability-topology.

**Comment by JonasAhl** [comment]
Bandwidth utilization is recommended to be based on "tx-bytes-sent".
"tx-bytes-sent is not included/supported in the RFC 8561 device model.
out-octets is included in RFC 7223, but is referring to the ethernet level.
Time slot for the measurement still remains to be decided - or should we allow vendor specific implementations, described in a separate data node for the time window used.

**Comment by JonasAhl** [comment]
Continued to discuss Issue #3:
Clarify that it is about utilized bandwidth, not utilization in percentage.
Change the name of the attribute accordingly.
Proposal is to add a node/leaf for the time-window.
Clarify/describe that the utilization can be calculated in different ways depending on the policy used. 
Compare e.g. with max in highest modulation, or max in current modulation, etc.

**Comment by italobusi** [comment]
Let's check that we have a common view about the definitions:

<img src="https://latex.codecogs.com/gif.image?\dpi{110}utilizedBw=\frac{txBytes}{timeWindow}*8"/>

<img src="https://latex.codecogs.com/gif.image?\dpi{110}BwUtilization=\frac{utilizedBw}{linkBw}"/>

In order to get comparable values, a common/standard definition is needed for
1. _txBytes_
2. _timeWindow_ 
3. _linkBw_

Reporting the _utilizedBw_ instead of the _BwUtilization_ removes the need for a common/standard definition of the _linkBw_.

Reporting the _timeWindow_ removes the need for a common/standard definition of the _timeWindow_,

I am not sure whether we have a common/standard definition for counting the _txBytes_: to be investigated.

**Comment by DanielaSpreafico** [comment]
I agree with the concept and the formula about utilizedBw and BwUtilization.

The txBytes has to be the bytes sent on the timeWindow, so the difference between the tx bytes sent at T0 and tx bytes sent at T1 where T1-T0 = timeWindow.

**Comment by JonasAhl** [comment]
I also agree with the concept and Daniela´s addition, but I'm not sure I understand the first formula.
Why is there a multiplication with 8?
Isn't 1 byte = 8 bits and not the opposite, which I think could be the mistake.

**Comment by italobusi** [comment]
> I also agree with the concept and Daniela´s addition, but I'm not sure I understand the first formula. Why is there a multiplication with 8? Isn't 1 byte = 8 bits and not the opposite, which I think could be the mistake.

Good point, I think that adding the measurements units would help:

<img src="https://latex.codecogs.com/gif.image?\dpi{110}utilizedBw=\frac{txBytes*8}{timeWindow}\;&space;[bits/sec]"/>

<img src="https://latex.codecogs.com/gif.image?\dpi{110}BwUtilization=\frac{utilizedBw}{linkBw}*100\,&space;[%]"/>

**Comment by samans** [comment]
A few observations and questions about the bandwidth topic.
[mw-bandwidth-questions.pptx](https://github.com/ietf-ccamp-wg/draft-ietf-ccamp-mw-topo-yang/files/9878722/mw-bandwidth-questions.pptx)
